### PR TITLE
drm_vo: monitor aspect from --monitoraspect or the physical screen size 

### PR DESCRIPTION
**Pull request by sgerwk**
I agree that my changes can be relicensed to LGPL 2.1 or later.

### CODE REVIEW

**Comment by sgerwk** [comment]
To have mpv print the physical screen size run it with option --msg-level=vo=v; a message is printed if the hardware does not provide this information.

File: video/out/vo_drm.c [hunk 1]
```diff
@@ -443,7 +445,24 @@ static int preinit(struct vo *vo)
         goto err;
     }
 
-    vo->monitor_par = 1 / vo->opts->monitor_pixel_aspect;
+    screenaspect = p->screen_w / (double) p->screen_h;
```
Comment (kevmitch) [inline video/out/vo_drm.c]: might as well declare it `double screenaspect` here.

File: DOCS/man/vo.rst [hunk 2]
```diff
@@ -492,6 +492,13 @@ Available video output drivers are:
         Overlay index is zero based, and related to crtc.
         (default: 0)
 
+    ``--drm-aspect``
```
Comment (kevmitch) [inline DOCS/man/vo.rst]: is this option really needed? can't it just use 

1) `--monitoraspect`
2) if not set, then `--monitorpixelaspect`
3) if not set, then use kms connector values

If for some reason the previous behaviour is really wanted, you can always set `--monitorpixelaxpect=1`.

**Comment by kevmitch** [comment]
commit message header should be shortened to fit on the line. maybe something like

"drm_vo: use --monitoraspect or the physical screen size"

**Comment by sgerwk** [comment]
Using the physical screen size by default requires changing the default value of monitor_pixel_aspect from 1 to 0,  as otherwise there would be no way to tell whether --monitorpixelaspect was given with value 1 or not given at all. The default 0 is IMO quite logical, and requires only an handful of changes to the other files (second last commit above).

With the new default, the code for using the physical screen size in vo_drm simplifies a lot.

**Comment by kevmitch** [comment]
I've been doing a bit more testing and looking into this. It seems that the physical dimensions are not quite exact resulting in aspects close to but no equal to 1 on my 1920x1080 LCD (square pixel) display. From the [log](https://0x0.st/sTwF.log), it says that it's choosing an aspect of 0.993464. Consequently, I see a few extra blank pixel columns on either side of a 16x9 video. 

I'm not sure where exactly these physical dimensions come from, but those provided by xrandr in x11 (344mm x 193mm) have a similar problem of implying a PAR of 0.997. I also note that if I change the xrandr resolution to a different aspect ratio like 1680x1050, the output is letter-boxed, rather than stretched to the screen width. This means that the PAR should still probably be something close to 1, however the reported physical dimensions remain unchanged from their ~16:9 values, incorrectly implying a non-square PAR for that resolution. I'm not sure how the equivalent would work with drm, but I can imagine it could easily run into the same problem.

So I think I owe you an apology for sending you on a wild goose chase. It's probably better to stick to the default of PAR=1. Do you still think there's value in reading the physical dimensions?

**Comment by sgerwk** [comment]
The mmWidth and mmHeight fields come from the monitor [EDID](Extended_Display_Identification_Data). They may give an hint of the size of the screen, but are apparently not precise enough for determining the monitor aspect. So, better stick to --monitoraspect and --monitorpixelaspect (last commit above).

For the report, when setting a mode that is not one of the original modes of the connector, the result depends on the hardware: one computer stretches the image to fill the screen; other two use one of the original modes and cropped or panned the image. AFAIK, setting a mode that is not one of the original modes of the connector is impossible in mpv, since --drm-mode only allows to specify a mode that is one of the original ones of the connector.

**Comment by kevmitch** [comment]
Go ahead and squash. Maybe clarify in the commit message that it still falls back on `--monitorpixelaspect` if `--monitoraspect` is unset.

**Comment by kevmitch** [comment]
merged in  5cc796daccd3f3ccc0506d4423fd52e00c12218a.
I added braces as per coding style. Sorry I missed this previously, but I didn't want to bother you about it.

**Pull request closed without merging**